### PR TITLE
Enable sending non-JSON formatted data in generic api requests

### PR DESCRIPTION
**Pull request by tlium**
Added a new command argument to servicenow-generic-api-call and modified the the generic_api_call_command function to handle the new argument. Sane default value and functionality ensures backwards compatibility.

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Enable seding non-JSON formatted data in servicenow-generic-api-call

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [X] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by tlium** [comment]
Do I need to squash the two commits before merging is considered?

**Comment by ostolero** [comment]
Hi @tlium , no need to squash the commits together, we'll do it when merging.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml [hunk 1]
```diff
@@ -2179,7 +2179,13 @@ script:
     - description: the API path starting with forward slash (/)
       name: path
       required: true
-    - description: json to send in body
+    - auto: PREDEFINED
+      defaultValue: 'true'
+      description: is the body json formatted
+      predefined:
+      - 'true'
+      - 'false'
```
Comment (ostolero) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml:2187]: add the field "name : json_body"

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 2]
```diff
@@ -2669,7 +2669,11 @@ def generic_api_call_command(client: Client, args: Dict) -> Union[str, CommandRe
     method = str(args.get("method"))
     path = str(args.get("path"))
     headers = json.loads(str(args.get("headers", {})))
-    body: Dict = json.loads(str(args.get("body", {})))
+    json_body = args.get("json_body", "true")
+    if json_body == "true":
+        body: Dict = json.loads(str(args.get("body", {})))
+    else:
+        body = args.get("body", "")
```
Comment (ostolero) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:2676]: What other option is there for the body? can the same functionality be achieved by this code? -
```suggestion
   try:
        body: Dict = json.loads(str(args.get("body", {})))
    except ValueError:
        body = args.get("body", "")
```

**Review by ostolero — CHANGES_REQUESTED** [review]
Hi @tlium , thanks for the contribution! the change looks good :)
See my comments in the code, also note these changes-
1. Add release notes, can be done using our sdk [update-rn](https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/update_release_notes/README.md) command.
2.  Update README file with the new argument, you can do it manually I think might be easier then using the sdk command.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py (hunk 2 above)
Comment (tlium) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:2676]: I suppose that's another way to do it. But shouldn't the  caught exception should be a JSONDecodeError?

**Review by ostolero — APPROVED** [review]
Everything looks great. demo video was done :)

**Pull request merged**